Add custom data to item in mesh library

Crashim03 opened the pull request:
Enable creation of custom data layers in mesh libraries
Co-authored-by: João Santos <joao.carvalho.santos@tecnico.ulisboa.pt>

Closes: godotengine/godot-proposals/issues/9995

<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

Crashim03 commented:
> You removed two test cases from `test_main`, they need to be restored, you didn't handle the conflicts correctly

Fixed it. Thank you for your patience

AThousandShips commented:
You might need to move it down with the other 3D only tests under `#ifndef _3D_DISABLED`, but that'll show up in the CI if it's needed

AThousandShips commented:
Realized this won't turn up in the CI so please move it as the grid map module isn't enabled with 3D disabled

AThousandShips commented:
Did you mean to close this? Looks like you accidentally reset your branch

alex-telo committed:
feat: add custom data to item in mesh library
Enable creation of custom data layers in mesh libraries
Co-authored-by: João Santos <joao.carvalho.santos@tecnico.ulisboa.pt>

Crashim03 commented:
> Did you mean to close this? Looks like you accidentally reset your branch

Yes it was a mistake. Moved the include under the `#ifndef _3D_DISABLED` and fixed a small issue.

AThousandShips commented:
The includes there need to be sorted

vhladko commented:
Any updates on this one ? It's really hard to maintain tiles when there is no way to add some types, groups, etc to specific items

StarmanAkremis commented:
Please add this

CenTdemeern1 commented:
@Crashim03 is there any chance that you could update the branch? I'd like to find and send this to someone on rocket chat in hopes of getting it reviewed and then hopefully merged later.
(Sorry if this counts as bumping the PR, I couldn't find any other way to contact the author)

btflylog commented:
@Crashim03 @AThousandShips 
Would it be okay to take over this PR? There’s little sense in letting the dust settle on a critical feature.

AThousandShips commented:
We do not accept feature PRs by new contributors [see here](https://contributing.godotengine.org/en/latest/pull_requests/pull_request_guidelines.html#no-features-from-new-contributors) so someone else would have to do that

DaveTheEggman commented:
> We do not accept feature PRs by new contributors [see here](https://contributing.godotengine.org/en/latest/pull_requests/pull_request_guidelines.html#no-features-from-new-contributors) so someone else would have to do that

Could I do it? I'll discuss it in the contributors chat

AThousandShips commented:
I'd suggest focusing on the PRs you have already and someone else can pick this up

CenTdemeern1 commented:
Would that someone else have to stumble upon this PR themselves?
